docs/editors/emacs.md

```diff
@@ -0,0 +1,124 @@
+---
+id: emacs
+title: Emacs
+---
+
+Metals works in Emacs thanks to the the
+[`lsp-scala`](https://github.com/rossabaker/lsp-scala) package.
+
+![Emacs demo](../assets/emacs-demo.gif)
+
+> The Emacs LSP client has [several known issues](#known-issues). Most
+> critically, diagnostics are not published for unopened buffers meaning compile
+> errors can get lost.
+
+```scala mdoc:requirements
+
+```
+
+## Installation
+
+First, clone the `lsp-scala` repository.
+
+```sh
+git clone https://github.com/rossabaker/lsp-scala.git
+```
+
+Next, update your Emacs configuration to load `lsp-scala` along with its
+dependencies
+
+```el
+;; Enable scala-mode and sbt-mode
+(use-package scala-mode
+  :interpreter
+  ("scala" . scala-mode))
+(use-package sbt-mode
+  :commands sbt-start sbt-command
+  :config
+  ;; WORKAROUND: https://github.com/ensime/emacs-sbt-mode/issues/31
+  ;; allows using SPACE when in the minibuffer
+  (substitute-key-definition
+   'minibuffer-complete-word
+   'self-insert-command
+   minibuffer-local-completion-map))
+
+;; Enable nice rendering of diagnostics like compile errors.
+(use-package flycheck
+  :ensure t
+  :init (global-flycheck-mode))
+(use-package lsp-ui)
+(require 'lsp-ui)
+
+(use-package lsp-mode)
+(add-hook 'lsp-mode-hook 'lsp-ui-mode )
+(require 'lsp-mode)
+(require 'sbt-mode)
+
+(add-to-list 'load-path "~/path/to/lsp-scala")
+(require 'lsp-scala)
+(add-hook 'scala-mode-hook #'lsp-scala-enable)
+```
+
+> You may need to disable other packages like `ensime` or sbt server to prevent
+> conflicts with Metals.
+
+If you start Emacs now then it will fail since the `metals-emacs` binary does
+not exist yet.
+
+```scala mdoc:bootstrap:metals-emacs emacs
+
+```
+
+The `-Dmetals.client=emacs` flag is important since it configures Metals for
+usage with Emacs.
+
+```scala mdoc:editor:emacs
+
+```
+
+## Manually trigger build import
+
+To manually trigger a build import, run `M-x lsp-scala-build-import`.
+
+![Import build command](assets/emacs-import-build-command.png)
+
+## Run doctor
+
+Run `M-x lsp-scala-doctor-run` to troubleshoot potential configuration problems
```

Comment:
This command does not exist yet in `lsp-scala`, it would be good to add it https://scalameta.org/metals/docs/editors/new-editor.html#run-doctor